Properly raise exception on connection problem

eyj opened the pull request:
We had a problem where a CancellationError was returned to a waiting request in case the redis died. This PR properly raises the right exception, and still retains the right behaviour on QUIT.

trezorg committed:
Redis cluster. Initial. Processing Moved command.

It is working only with masters nodes properly.

trezorg committed:
Fixed cluster initialization

Commit:
Added IntelliJ config file to gitignore.

Commit:
Tests now run in a sentinel setup.

Commit:
Added a script to setup an MMM SSS cluster.

Commit:
Started to make the old tests work against a cluster. Created runclustertests.py. Fixed a bug in key_slot.

trezorg committed:
Fixed key_slot method. Force bytes

Commit:
More tests work on a Redis cluster.

Commit:
HLL and sorted set tests work with cluster.

Commit:
Converted remaining tests to run on a cluster. Except those where functionality is not implemented yet or doesn't make sense on a cluster.

Commit:
Fixed SLOT_ZERO_KEY

Commit:
Added some tests for RedisCluster and RedisPoolCluster.

Commit:
Moved test cluster setup code into own package within aioredis so that it can be imported in application tests.

Commit:
Added reload_cluster_pool. Added methods to TestCluster to trigger failover.

Commit:
Implemented own cluster configuration script (using sockets). No need anymore to download the redis-trib.rb file from Redis source.

Commit:
Improved testcluster code: Better handling of error responses and retries.

Commit:
Improved socket.recv usage.

Commit:
Make setupcluster print a message that explains how to run the tests.

Commit:
Added a docstring.

eyj committed:
Properly raise exception on connection problem

nasenbaer13 commented:
Any plans to merge this pr? We would also like to see this in https://github.com/aio-libs/aioredis. So @trezorg do you want to create a pull request or shall we create one from our fork?

Commit:
Prepare cluster pool parallelized

Commit:
Reset cluster pool after reloading cluster pool not before to prevent race conditions when accessing the cluster_pool

Commit:
Access node via its unique id and not by its position in the node list

Commit:
RedisCluster asks all nodes for cluster info and takes the first returned info. Various improvements.

Commit:
Really parallelized get_cluster_pool.

Commit:
Fixed tests after merge.

Commit:
Removed broken iscan method.

Commit:
Fixed _read_bulk_string_response.

Commit:
Improved cluster examples. Now they can be used together with the setupcluster.py script.

Commit:
Improvements in cluster code.

Commit:
Small improvements.

Commit:
Added tests for wait_closed fix.

Commit:
Comply with PEP8.

Commit:
Fixed issues after rebasing.

Commit:
Fixed flake8 warnings.

Commit:
Make travis start a cluster before running cluster examples.

Commit:
Fixed reload_cluster_pool and corresponding test.

Commit:
Make sure fetch cluster info connections are closed (to avoid Pending Task Destroyed warnings).

Commit:
Modified tests so that the runtests script can run both the standalone tests and the cluster tests and measure coverage appropriately.

Commit:
Only run cluster tests on travis, if Redis 3 is used.

Commit:
Hopefully fixed travis.yml file.

Commit:
Fixed example filter in travis script.

Commit:
More fixes in Travis script.

Commit:
Setup cluster using the correct version of Redis.

Commit:
More fixes for Travis.

Commit:
Make cluster examples run even if database is not empty.

Commit:
Fixed EVAL handling on cluster.

Commit:
Removed two unnecessary lines.

Commit:
Removed unused method.

eyj committed:
Properly clean up a pool if cancellation occurs during setup

Commit:
Added test for the last commit to keep coverage high.

Commit:
Handle new CLUSTER NODES format in Redis v4.

lemurchik committed:
Merge remote-tracking branch 'aio-libs/aioredis/master'

# Conflicts:
#	.gitignore
#	.travis.yml
#	aioredis/__init__.py
#	aioredis/connection.py
#	aioredis/errors.py
#	aioredis/pool.py
#	aioredis/util.py
#	runtests.py
#	tests/_testutil.py
#	tests/connection_commands_test.py
#	tests/connection_test.py
#	tests/generic_commands_test.py
#	tests/hash_commands_test.py
#	tests/hyperloglog_commands_test.py
#	tests/list_commands_test.py
#	tests/pool_test.py
#	tests/pubsub_commands_test.py
#	tests/reconnect_test.py
#	tests/scripting_commands_test.py
#	tests/server_commands_test.py
#	tests/set_commands_test.py
#	tests/sorted_set_commands_test.py
#	tests/string_commands_test.py
#	tests/transaction_commands_test.py

lemurchik committed:
- Code changes for compatibility with aioredis 1.0.0
- Changing code for python >= 3.5
- Finishing all cluster management commands
- Tests on py.tests for maximum code coverage
- Fixing some errors
- Cluster management commands examples

lemurchik committed:
Me as contributor

lemurchik committed:
Fixing examples

lemurchik committed:
Fixing examples, again.

lemurchik committed:
- Typos fix
- Variables naming change
- Functionality improvements
- Flake8
- Tests improvements

lemurchik committed:
- Removing unused functionality
- Additional test for coverage improvement
- Code improvements

The-Panda committed:
Merge pull request #2 from lemurchik/master

Finishing functionality for redis cluster

Commit:
Fix test_del_slots: Also send del command to slaves.

Commit:
Fix flake8 config.

Commit:
Fix two typos.

Commit:
Make get_nodes_entities private. Make failover test more robust. Write some tests for normal commands on cluster.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Loading cluster info may fail with an OSError.

Commit:
Use pytest tmpdir fixture to create temporary cluster configuration directory.

Commit:
Silence error when temporary directories cannot be deleted.

Commit:
Try to get rid of strange errors in AppVeyor on Windows.

Commit:
Fix test_add_slots which would sometimes fail in AppVeyor.

Commit:
Skip cluster tests when using a Redis version < 3.

Commit:
Delete old examples

Commit:
Try to setup cluster when running examples on Travis.

Commit:
Remove non-ASCII chars from setup.cfg, or distutils will crash.

Commit:
Exclude testcluster file from coverage. This module is for testing purposes and not tested itself.

Commit:
Improve _wait_result function in tests.

Commit:
Remove support for CLUSTER SETSLOTS, because test does not work reliably.

Commit:
Try to avoid timeout in test_del_slots.

Commit:
Fix setupcluster.py script.

Commit:
Remove unused setup_test_cluster functionality.

popravich committed:
few small amendments in sentinel

fuchaoqun committed:
reduce encode_command cost about 60%+

fuchaoqun committed:
fix flake8 error

popravich committed:
Merge pull request #397 from fuchaoqun/master

reduce encode_command cost about 60%+

alanc10n committed:
Fix zcount grater and then typos

alanc10n committed:
Fix then/than typos

popravich committed:
Merge pull request #400 from alanc10n/typo-fixes

Typo fixes

filmor committed:
Fix INFO parsing

Nothing prevents the server from returning values that contain a colon.

popravich committed:
update PyPy3 version to v6.0.0

mmniaziqb committed:
Merge branch 'master' of https://github.com/aio-libs/aioredis into orig

mmniaziqb committed:
minor refactoring

mmniaziqb committed:
fix misleading logs, because of generator already consumed

mmniaziqb committed:
only execute on all nodes if explicitly asked for, otherwise execute on random node

popravich committed:
Merge pull request #405 from filmor/patch-1

Fix INFO parsing

mmniaziqb committed:
Merge branch 'master' into merged

The-Panda committed:
Merge pull request #3 from mmniaziqb/master

Merge latest master

Commit:
Fix tests after making 'many' parameter required.
